opamp/observiq/updater_manager_windows.go

```diff
@@ -0,0 +1,71 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package observiq
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// Ensure interface is satisfied
+var _ UpdaterManager = (*WindowsUpdaterManager)(nil)
+
+// WindowsUpdaterManager handles starting a Updater binary and watching it for failure with a timeout
+type WindowsUpdaterManager struct {
+	tmpPath string
+	logger  *zap.Logger
+}
+
+// newUpdaterManager creates a new UpdaterManager
+func newUpdaterManager(defaultLogger *zap.Logger, tmpPath string) UpdaterManager {
+	return &WindowsUpdaterManager{
+		tmpPath: filepath.Clean(tmpPath),
+		logger:  defaultLogger.Named("updater manager"),
+	}
+}
+
+// StartAndMonitorUpdater will start the Updater binary and wait to see if it finishes unexpectedly.
+// While waiting for Updater, it should kill the collector and we should never execute any code past running it
+func (m WindowsUpdaterManager) StartAndMonitorUpdater() error {
+	updaterPath := filepath.Join(m.tmpPath, updaterDir, updaterName)
+	absTmpPath, err := filepath.Abs(m.tmpPath)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path of tmp dir: %w", err)
+	}
+	cmd := exec.Command(updaterPath, "--tmpdir", absTmpPath)
```

Comment:
I think we only do gosec in CI on linux right now, but may as well add the nosec here too:
```suggestion
        //#nosec G204 -- paths are not determined via user input
	cmd := exec.Command(updaterPath, "--tmpdir", absTmpPath)
```

Reply:
@BinaryFissionGames changed